sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/jit/ToJitInnerVisitor.java

```diff
@@ -360,11 +360,16 @@ public JITBlock getCurrentBlock() {
      * @param name           Name of function to call.
      * @param expression     Operations which is being translated.
      *                       Usually an ApplyExpression, but not always.
+     * @param resultType     Expected result type.  If null, type obtained from expression.
      * @param arguments      Arguments to supply to the function call.
      */
-    void createFunctionCall(String name,
+    JITInstructionPair createFunctionCall(String name,
+                            @Nullable DBSPType resultType,
```

Comment:
Feels cleaner to have `createFunctionCall(name, resultType, expression, arguments)` and `createFunctionCall(name, expression, arguments)`. The latter simply does what you do here and calls `createFunctionCall(name, expression.getType(), expression, arguments)`, instead of having `null` be an argument at several call sites.